src/modules/user/user.repository.ts

```diff
@@ -0,0 +1,24 @@
+import { PrismaClient, User } from '@prisma/client';
+const prisma = new PrismaClient();
+export interface UserRepository {
+  findByEmail(email: string): Promise<User | null>;
+  findById(id: number): Promise<User | null>;
+  updatePassword(id: number, newPassword: string): Promise<User | null>;
+}
+
+export class UserRepositoryImpl implements UserRepository {
```

Comment:
Missing @Injectible() decorator, when using prisma as the repository "provider", the repository itself is a service, so please fix this.